app/src/main/java/com/infomaniak/drive/ui/SaveExternalFilesActivity.kt

```diff
@@ -91,6 +93,14 @@ class SaveExternalFilesActivity : BaseActivity() {
         fetchSelectedDrive()
         fetchFolder()
         setupSaveButton()
+
+        fileNameEdit.selectAllButFileExtension()
+    }
+
+    private fun TextInputEditText.selectAllButFileExtension() {
+        val fileName = (text ?: "").toString()
+        val endIndex = fileName.substringBeforeLast(".").length
+        post { setSelection(0, endIndex) }
```

Comment:
It is necessary to merge the logic with this method
https://github.com/Infomaniak/android-kDrive/blob/master/app/src/main/java/com/infomaniak/drive/data/models/File.kt#L201